lm_eval/models/huggingface.py

```diff
@@ -678,39 +678,61 @@ def _create_tokenizer(
             )
         return None
 
-    def _detect_batch_size(self, requests=None, pos: int = 0):
-        if requests:
+    def _detect_batch_size(self, requests=None, pos: int = 0) -> int:
+        if len(requests[0]) == 3:  # logprob evals
             _, context_enc, continuation_enc = requests[pos]
             max_length = len(
                 (context_enc + continuation_enc)[-(self.max_length + 1) :][:-1]
             )
             max_context_enc = len(context_enc[-(self.max_length + 1) :])
             max_cont_enc = len(continuation_enc[-(self.max_length + 1) :])
-        else:
-            max_length = self.max_length
+            security_margin_factor = (
+                4  # batch sizes for log prob evals sometimes generate OOMs
```

Comment:
Seems like some sort of magic number duplicated in several places. May I suggest making it constant at the beginnig of the file and using this constant if needed

Reply:
There would need to be 2 constants, as it is often loglikelihood requests that OOM and we would like to optimize batch size as much as possible